pkg/syncer/upsync/upsync_reconcile.go

```diff
@@ -0,0 +1,261 @@
+		if err != nil {
+			return false, err
+		}
+		preparedResource.SetResourceVersion(existingResource.GetResourceVersion())
+		updatedResource, err := c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).Update(ctx, preparedResource, metav1.UpdateOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		// - update the status as a distinct action,
+		preparedResource.SetResourceVersion(updatedResource.GetResourceVersion())
+		updatedResource, err = c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).UpdateStatus(ctx, preparedResource, metav1.UpdateOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		// - finally update the main content again to set the resource version annotation to the value of the downstream resource version.
+		preparedResource.SetAnnotations(addResourceVersionAnnotation(resourceVersionDownstream, preparedResource.GetAnnotations()))
+		preparedResource.SetResourceVersion(updatedResource.GetResourceVersion())
+		_, err = c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).Update(ctx, preparedResource, metav1.UpdateOptions{})
+		// If the downstream resource is marked for deletion, let's requeue it to manage the deletion timestamp
+		return markedForDeletionDownstream, err
+	}
+
+	if downstreamResource.GetDeletionTimestamp() != nil {
+		return false, c.removeUpstreamResource(ctx, gvr, upstreamClusterName, upstreamNamespace, upstreamName, false)
+	}
+
+	return false, nil
+}
+
+func (c *controller) prepareResourceForUpstream(ctx context.Context, gvr schema.GroupVersionResource, upstreamNS string, upstreamLogicalCluster logicalcluster.Name, downstreamObj *unstructured.Unstructured) *unstructured.Unstructured {
+	// Make a deepcopy
+	resourceToUpsync := downstreamObj.DeepCopy()
+	annotations := resourceToUpsync.GetAnnotations()
+	if annotations != nil {
+		delete(annotations, shared.NamespaceLocatorAnnotation)
+		resourceToUpsync.SetAnnotations(annotations)
+	}
+	labels := resourceToUpsync.GetLabels()
+	if labels != nil {
+		delete(labels, workloadv1alpha1.InternalDownstreamClusterLabel)
+		resourceToUpsync.SetLabels(labels)
+	}
+	resourceToUpsync.SetNamespace(upstreamNS)
+	resourceToUpsync.SetUID("")
+	resourceToUpsync.SetResourceVersion("")
+	resourceToUpsync.SetManagedFields(nil)
+	resourceToUpsync.SetDeletionTimestamp(nil)
+	resourceToUpsync.SetDeletionGracePeriodSeconds(nil)
+	resourceToUpsync.SetOwnerReferences(nil)
+	resourceToUpsync.SetFinalizers([]string{shared.SyncerFinalizerNamePrefix + c.syncTargetKey})
+
+	return resourceToUpsync
+}
+
+func (c *controller) removeUpstreamResource(ctx context.Context, gvr schema.GroupVersionResource, clusterName logicalcluster.Name, namespace, name string, force bool) error {
```

Comment:
deleteUpstreamResource ? delete is the term in kube.

Reply:
Fixed in commit fffff6927c99c86702c7709881a77057331950cd